shared/src/ledger/native_vp.rs

```diff
@@ -241,3 +268,78 @@ where
         }
     }
 }
+
+impl<'a, DB, H> ClientReader for Ctx<'a, DB, H>
```

Comment:
I think we could move this impl to the ibc module, so that none of the ibc imports are here.